### DISCUSSION TITLE
DATA_ROOT/app is not created automatically.

### DISCUSSION THREAD

**Discussion by pfw** [piku/piku#328] [Q&A] [answered]
Is there a particular reason that a directory isn't created in DATA_ROOT for apps automatically? The other directories like LOG_ROOT/app are created.

**Comment by rcarmo** [comment]
No particular reason. Most applications tend to store data in a database of some sort (which may reside someplace else altogether), so that hasn't really been necessary until now. 

Feel free to do a PR, should be a 2-line thing (create and remove the data when the app is gone).

**Comment by rcarmo** [comment]
Actually, I went and checked and yes, `DATA_ROOT` is created automatically (I ended up adding it around the time I added support for `nginx` caching):

https://github.com/piku/piku/blob/master/piku.py#L1518

Are you using an older version?

**Comment by rcarmo** [comment, marked as answer]
Sorry, you were right. I just added that, since I now realise what you meant - I forgot to merge those 3 lines from my development branch. :)

**Reply by pfw** [reply]
Good one, thanks! It wasn’t a big deal at all but seemed odd. I’m very much enjoying the straightforwardness of piku.
